### PR TITLE
[Design Yaml] Add labels and annotations to root and clean final yaml

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
- [add labels and annotations in the root level of pattern-data](https://github.com/meshery/meshery/commit/9a6662d4a7f93bb47cc9443e108a87fe38ddb2d3) 
- clean redudant data from settings spec, since they are present in the root level

The elements strigified json: 
```
 "name: new Design\nservices:\n  application:\n    name: application\n    type: Application\n    apiVersion: core.oam.dev/v1alpha1\n    namespace: default\n    labels:\n      hello: hellovalue\n    settings:\n      advanced:\n        create_service: true\n      replicas: 3\n    traits:\n      meshmap:\n        edges: []\n        id: c8015278-101f-4d98-b65a-324182bf4caa\n        label: application\n        position:\n          posX: 165.76236541203224\n          posY: 114.84364277738584\n"
```

will result into final Yaml that will look like:
```
name: new Design 
services: 
  application: 
    name: application 
    type: Application 
    apiVersion: core.oam.dev/v1alpha1 
    namespace: default 
    labels: 
      hello: hellovalue 
    settings: 
      advanced: 
        create_service: true 
      replicas: 3 
    traits: 
      meshmap: 
        edges: [] 
        id: c8015278-101f-4d98-b65a-324182bf4caa 
        label: application 
        position: 
          posX: 165.76236541203224 
          posY: 114.84364277738584 
```

### Notes for reviewer

name, namespace, labels and annotations aren't the part of settings now.
Now settings will only contain spec values

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: server/models/pattern/core/pattern.go [hunk 1]
```diff
@@ -255,6 +284,13 @@ func NewPatternFileFromCytoscapeJSJSON(name string, byt []byte) (Pattern, error)
 		}
 		eleToSvc[ele.Data.ID] = svc.Name //will be used while adding depends-on
 		pf.Services[svc.Name] = &svc
+
+		// clean some redundant fields from the settings
+		delete(svc.Settings, "name")
+		delete(svc.Settings, "namespace")
```
Comment (Revolyssup) [inline server/models/pattern/core/pattern.go:290]: Why to put these here in the first place?

**Review by Revolyssup — CHANGES_REQUESTED** [review]

File: server/models/pattern/core/pattern.go (hunk 1 above)
Comment (Revolyssup) [reply server/models/pattern/core/pattern.go:290]: Lets not add these fields inside of `settings`

Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go:290]: remove redudancy.
the name, namespace, labels and annotations were in two places with the exact same value being duplicated.

Comment (Abhishek-kumar09) [reply server/models/pattern/core/pattern.go:290]: I may have get your next message after writing my first.

**Comment by leecalcote** [comment]
Ready for merge

**Comment by Abhishek-kumar09** [comment]
Btw, this was only there on conversion step that was being used by MeshMap. Anyways, I have created a PR in meshery-extensions. @Revolyssup You need to make some changes in Server to accommodate it.

**Pull request closed without merging**